includes/abstracts/abstract.llms.database.query.php

```diff
@@ -235,220 +119,56 @@ protected function get_skip() {
-			return;
-		}
-
-		foreach ( (array) $this->get( 'sort' ) as $orderby => $order ) {
-
-			if ( ! in_array( $orderby, $this->allowed_sort_fields, true ) || ! in_array( $order, array( 'ASC', 'DESC' ), true ) ) {
-
-				unset( $this->arguments['sort'][ $orderby ] );
-
-			}
-		}
-
-	}
-
-	/**
-	 * Sets a query variable.
-	 *
-	 * @since 3.8.0
-	 *
-	 * @param string $key Variable key.
-	 * @param mixed  $val Variable value.
-	 * @return void
-	 */
-	public function set( $key, $val ) {
-		$this->query_vars[ $key ] = $val;
-	}
-
 	/**
 	 * Set variables related to total number of results and pages possible with supplied arguments.
 	 *
 	 * @since 3.8.0
 	 * @since 4.5.1 Bail early if the query arg `no_found_rows` is true, b/c no reason to calculate anything.
+	 * @deprecated [version] `LLMS_Database_Query::set_found_results()` is deprecated.
 	 *
 	 * @return void
 	 */
 	protected function set_found_results() {
 
-		global $wpdb;
+		_deprecated_function( 'LLMS_Database_Query::set_found_results()', '[version]' );
 
 		// If no results, or found rows not required, bail early b/c no reason to calculate anything.
 		if ( ! $this->number_results || $this->get( 'no_found_rows' ) ) {
 			return;
 		}
 
-		$this->found_results = absint( $wpdb->get_var( 'SELECT FOUND_ROWS()' ) ); // db call ok; no-cache ok.
+		$this->found_results = $this->found_results();
 		$this->max_pages     = absint( ceil( $this->found_results / $this->get( 'per_page' ) ) );
 
 	}
 
 	/**
-	 * Setup arguments prior to a query.
+	 * Perform a SQL to retrieve the total number of found results for the given query.
 	 *
-	 * @since 3.8.0
-	 * @since 3.34.0 Sanitizes sort parameters.
-	 * @since 4.5.1 Added filter `"llms_{$this->id}_query_parse_args"`.
+	 * @since [version]
 	 *
-	 * @return void
+	 * @return int
 	 */
-	protected function setup_args() {
-
-		$this->arguments = wp_parse_args( $this->arguments_original, $this->arguments_default );
-
-		$this->parse_args();
-
-		if ( ! $this->get( 'suppress_filters' ) ) {
-			/**
-			 * Filters the parsed query arguments.
-			 *
-			 * The dynamic part of the filter `$this->id` identifies the extending query.
-			 *
-			 * @since 4.5.1
-			 *
-			 * @param array               $ars           The query parse arguments.
-			 * @param LLMS_Database_Query $db_query      The LLMS_Database_Query instance.
-			 * @param array               $original_args Original arguments before merging with defaults.
-			 * @param array               $default_args  Default arguments before merging with original.
-			 */
-			$this->arguments = apply_filters( "llms_{$this->id}_query_parse_args", $this->arguments, $this, $this->arguments_original, $this->arguments_default );
-		}
-
-		foreach ( $this->arguments as $arg => $val ) {
+	protected function found_results() {
 
-			$this->set( $arg, $val );
-
-		}
-
-		$this->sanitize_sort();
+		global $wpdb;
+		return absint( $wpdb->get_var( 'SELECT FOUND_ROWS()' ) ); // db call ok; no-cache ok.
```

Comment:
```suggestion
		return (int) $wpdb->get_var( 'SELECT FOUND_ROWS()' ); // db call ok; no-cache ok.
```
can't we just cast this to int? Are we worried about negative values?

Reply:
We have different habits. I feel this *barely* matters but I'll do as you suggest because I don't feel strong enough about it one way or another.